src/Map.ts

```diff
@@ -0,0 +1,160 @@
+import { hasClass } from './decorators';
+import { is } from './object';
+import global from './global';
+
+export default class Map<K, V> {
+	private _keys: K[] = [];
```

Comment:
Change to `protected` so that if someone wants to make an extension, it can access these